lib/src/main/java/org/exoplatform/social/addons/storage/entity/ProfileEntity.java

```diff
@@ -0,0 +1,165 @@
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.exoplatform.social.addons.storage.entity;
+
+import org.exoplatform.commons.api.persistence.ExoEntity;
+
+import javax.persistence.CollectionTable;
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.Lob;
+import javax.persistence.MapKeyColumn;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author <a href="mailto:tuyennt@exoplatform.com">Tuyen Nguyen The</a>.
+ */
+@Entity
+@ExoEntity
+@Table(name = "SOC_IDENTITY_PROFILE")
+@NamedQueries({
+        @NamedQuery(
+                name = "SocProfile.findByIdentity",
+                query = "SELECT p FROM ProfileEntity p WHERE p.identity.id = :identityId"
+        )
+})
+public class ProfileEntity {
+  @Id
+  @SequenceGenerator(name="SEQ_SOC_IDENTITY_PROFILE_ID", sequenceName="SEQ_SOC_IDENTITY_PROFILE_ID")
+  @GeneratedValue(strategy= GenerationType.AUTO, generator="SEQ_SOC_IDENTITY_PROFILE_ID")
+  @Column(name="PROFILE_ID")
+  private long id;
+
+  @OneToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "IDENTITY_ID")
+  private IdentityEntity identity;
+
+  private String url;
+
+  @Column(name = "AVATAR_URL")
+  private String avatarURL;
+
+  @Column(name = "AVATAR_MIMTYPE")
+  private String avatarMimeType;
+
+  @Lob
+  @Column(name = "AVATAR_IMAGE")
+  private byte[] avatarImage;
+
+  @ElementCollection(fetch = FetchType.EAGER)
+  @MapKeyColumn(name = "NAME")
+  @Column(name = "VALUE")
+  @CollectionTable(name = "SOC_IDENTITY_PROFILE_PROPERTY", joinColumns = {@JoinColumn(name = "PROFILE_ID")})
+  private Map<String, String> properties = new HashMap<String, String>();
+
+  @ElementCollection(fetch = FetchType.LAZY)
+  @CollectionTable(name = "SOC_IDENTITY_PROFILE_EXPERIENCE", joinColumns = {@JoinColumn(name = "PROFILE_ID")})
+  private List<ProfileExperience> experiences = new ArrayList<>();
+
+  @Column(name = "CREATED_TIME")
+  private long                      createdTime;
```

Comment:
So the bad thing here is that the Profile model class uses a long for a date. We must not align with this bad practice IMO but instead using Date in the database, even if it means to convert 2 times to go from data to UI. We must have a consistent data model. And in the future we will fix the Profile model class by switching to a date also.